vcd/resource_vcd_vm_placement_policy.go

```diff
@@ -150,30 +157,45 @@ func sharedVcdVmPlacementPolicyRead(ctx context.Context, d *schema.ResourceData,
 		if policyName == "" {
 			return diag.Errorf("both Placement Policy name and ID are empty")
 		}
-		if pVdcId == "" {
-			return diag.Errorf("both Provider VDC ID and Placement Policy ID are empty")
-		}
-
 		method = "name"
 		queryParams := url.Values{}
-		queryParams.Add("filter", fmt.Sprintf("name==%s;pvdcId==%s", policyName, pVdcId))
-		filteredPoliciesByName, err := vcdClient.GetAllVdcComputePoliciesV2(queryParams)
+
+		var foundPolicies []*govcd.VdcComputePolicyV2
+		var err error
+		// Fetches the VM Placement Policy with Provider VDC information, intended for System admins
+		if vdcId == "" {
```

Comment:
Just for clarity and readability - could it also check that `pvdcId != ""`.
I know we are not expecting it here, but I think this aids readability

Reply:
Added!